authentik/brands/models.py

```diff
@@ -0,0 +1,94 @@
+"""brand models"""
+from uuid import uuid4
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from rest_framework.serializers import Serializer
+from structlog.stdlib import get_logger
+
+from authentik.crypto.models import CertificateKeyPair
+from authentik.flows.models import Flow
+from authentik.lib.models import SerializerModel
+from authentik.lib.utils.time import timedelta_string_validator
+
+LOGGER = get_logger()
+
+
+class Brand(SerializerModel):
+    """Single brand"""
+
+    brand_uuid = models.UUIDField(primary_key=True, editable=False, default=uuid4)
+    domain = models.TextField(
```

Comment:
Do we need to do any additional checking here that this is globally unique across all tenants?

Reply:
Fortunately, no! As the tenant middleware is called before the brand middleware, even if the domain matches another tenant, the user will end up in the other tenant. So this domain has a lower priority than the one set in tenants.
(They also have different semantics, as the one in tenants is a perfect match whereas the one in brands is a suffix match)